### PR TITLE
docs: add release notes for 0.22.0

### PR DESCRIPTION
**Pull request by hanyucui**
Add release notes for 0.22.0.

### CODE REVIEW

**Review by tara-det-ai — APPROVED** [review]
fixed some capitalization issues

File: docs/release-notes.rst [hunk 1]
```diff
@@ -6,6 +6,68 @@
...
+
+**Release Date:** May 05, 2023
+
+**Breaking Change**
+
+-  The previous template CRUD endpoints have been removed from the `/templates/*` location. Please
+   use the APIs found at `/api/v1/templates/*`.
+
+-  Experiment: Optimizer has to be an instance of tensorflow.keras.optimizers.legacy.Optimizer
+   starting from Keras 2.11
```
Comment (tara-det-ai) [inline docs/release-notes.rst:24]: (did you mean "must be" ?)

Experiment: Optimizer must be an instance of ....

File: docs/release-notes.rst [hunk 2]
```diff
@@ -6,6 +6,68 @@
...
+         <https://github.com/tensorflow/tensorflow/releases/tag/v2.11.0>`_ for more details.
+
+      -  PyTorch users and users who specify custom images should not be affected.
+
+**Deprecated Features**
+
+-  Legacy TensorFlow 1 + PyTorch 1.7 + CUDA 10.2 support is deprecated and will be removed in a
+   future version. The final TensorFlow 1.15.5 patch was released in January 2021, and no further
+   security patches are planned. Consequently, we recommend users migrate to modern versions
+   TensorFlow 2.0 and PyTorch. Our default environment images currently ship with
```
Comment (drh-determined-ai) [inline docs/release-notes.rst:41]: I suggest writing "TensorFlow 2" here, since we're well beyond 2.0.

**Review by vishnu2kmohan — APPROVED** [review]
LGTM, thanks @hanyucui :rocket:

**Pull request merged**